packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -123,3 +106,57 @@ const useMenuSelectableState = (state: MenuState) => {
     });
   });
 };
+
+const useMenuOpenState = (state: MenuState) => {
+  const shouldHandleKeyboadRef = React.useRef(false);
+  const onOpenChange: MenuState['onOpenChange'] = useEventCallback((e, data) => state.onOpenChange?.(e, data));
+
+  const [open, setOpen] = useControllableValue(state.open, state.defaultOpen);
+  // TODO fix useControllableValue typing
+  state.open = open !== undefined ? open : state.open;
+
+  state.setOpen = React.useCallback(
+    (e, data) => {
+      setOpen(prevOpen => {
+        // More than one event (mouse, focus, keyboard) can request the popup to close
+        // We assume the first event is the correct one
+        if (prevOpen !== data.open) {
+          onOpenChange?.(e, { ...data });
+        }
+
+        if (data.keyboard) {
+          shouldHandleKeyboadRef.current = true;
+        }
+
+        return data.open;
+      });
+    },
+    [setOpen, onOpenChange],
+  );
+
+  // Manage focus for open state
+  const { findFirstFocusable } = useFocusFinders();
+  const focusFirstMenuItem = React.useCallback(() => {
+    const firstFocusable = findFirstFocusable(state.menuPopupRef.current);
+    firstFocusable?.focus();
+  }, [findFirstFocusable, state.menuPopupRef]);
+  React.useEffect(() => {
+    if (!shouldHandleKeyboadRef.current) {
+      return;
+    }
+
+    if (open) {
+      focusFirstMenuItem();
+    } else {
+      state.triggerRef.current?.focus();
+    }
+
+    shouldHandleKeyboadRef.current = false;
+  }, [state.triggerRef, shouldHandleKeyboadRef, focusFirstMenuItem, open, state.isSubmenu]);
```

Comment:
```suggestion
  }, [state.triggerRef, shouldHandleKeyboadRef, focusFirstMenuItem, open]);
```

Is `state.isSubmenu` is a dependency there?